sysrib/server_test.go

```diff
@@ -627,33 +681,102 @@ func TestServer(t *testing.T) {
 
 	for _, tt := range tests {
 		t.Run(tt.desc, func(t *testing.T) {
+			grpcServer := grpc.NewServer()
+			_, err := gnmi.New(grpcServer, "local")
+			if err != nil {
+				t.Fatal(err)
+			}
+			lis, err := net.Listen("tcp", "localhost:0")
+			if err != nil {
+				t.Fatalf("Failed to start listener: %v", err)
+			}
+			go func() {
+				grpcServer.Serve(lis)
+			}()
+
 			dp := NewFakeDataplane()
-			s, err := NewServer(dp)
+			s, err := NewServer(dp, lis.Addr().String(), "local")
 			if err != nil {
 				t.Fatal(err)
 			}
 
-			for _, intf := range tt.inInterfaces {
-				if err := s.addInterfacePrefix(intf.name, intf.ifindex, intf.prefix, intf.niName); err != nil {
-					t.Fatal(err)
+			clientCtx, cancel := context.WithCancel(context.Background())
+			var sendErr, recvErr error
+			go func(ctx context.Context) {
```

Comment:
why do this in gorouting if you blocking waiting to goroutine to finish right after?

Reply:
Removed.